internal/envoy/listener.go

```diff
@@ -66,7 +67,8 @@ func Listener(name, address string, port int, lf []*envoy_api_v2_listener.Listen
 
 // HTTPConnectionManager creates a new HTTP Connection Manager filter
 // for the supplied route and access log.
-func HTTPConnectionManager(routename string, accesslogger []*accesslog.AccessLog) *envoy_api_v2_listener.Filter {
+func HTTPConsnectionManager(routename string, accesslogger []*accesslog.AccessLog, requestTimeout *duration.Duration) *envoy_api_v2_listener.Filter {
```

Comment:
I'm sorry to go back and forth on this. this is just a minor complaint but it looks like there are no code paths where requestTimeout is nil so there can never be a case where the requestTimeout we pass to Envoy is nil -- its always explicitly set to a &proto.Duration{}. 

If that's the case then it is probably going to be lighter on the page to pass in a time.Duration to HTTPConnection manager. 
```
func HTTPConnectionManager(routename string, accesslogger []*accesslog.AccessLog, requestTimeout time.Duration) *envoy_api_v2_listener.Filter
```

Reply:
That makes sense to me. I'll change like follows:
- time.Duration to HTTPConnectionManager.
- HTTPConnectionManager will turn that into a duration.Duration (either 0 or >0).

Reply:
In the end, I put the validation logic into the requestTimeout method on the ListenerVisitorConfig, and moved HTTPConnection manager to time.Duration.